core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -197,6 +203,25 @@ sealed trait Cause[+E] extends Product with Serializable { self =>
                 Failure("An error was rethrown with a new trace." :: renderTrace(Some(trace))) :: causeToSequential(c).all
               )
           }
+        case Meta(cause, data) =>
+          if (data.stackless) cause match {
```

Comment:
You may want to pass `Data` in from above, that way you can have a default for all `Meta`, and each level can override it. For example, you might have `stack` nested inside `stackless` .That means you want the stack on the inside part but not on the outside part. This logic requires top-down inheritance of parent settings, with child override.

Reply:
I think we could do that by changing the last case to:

```scala
case m @ Meta(_, _) => causeToSequential(m)
```

The cases for all the other constructors that allow nesting already pass down the parent `Meta`. So then if you had nested `Meta` cases you would get to the `Meta(Meta)` branch and you would then trigger the rule that innermost scope takes precedence.

Reply:
I'm not convinced you can do it without a stack (either real stack or fake stack), as you have "edges" that should inherit parent behavior and "wedges" that have their own behavior, and you have to flip flop between them. However, we can fix this later, I am happy to get this merged in for 1.0.

Reply:
Yes, we are definitely simplifying a little bit here in that we are throwing away the second branch of any `Both` or `Then` cases, which might themselves contain metadata indicating that they should be shown. But as you say we can clean this up later,